core-it-suite/src/test/java/org/apache/maven/it/MavenIT0199CyclicImportScopeTest.java

```diff
@@ -0,0 +1,56 @@
+package org.apache.maven.it;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.File;
+import org.apache.maven.it.util.ResourceExtractor;
+
+public class MavenIT0199CyclicImportScopeTest extends AbstractMavenIntegrationTestCase {
+
+    public MavenIT0199CyclicImportScopeTest() {
+        super(ALL_MAVEN_VERSIONS);
```

Comment:
Well, tested against 3.5.2. Did not try running against, say, 2.x. Is there some automated process which does this?